owncloudApp/src/main/java/com/owncloud/android/ui/dialog/FileAlreadyExistsDialog.kt

```diff
@@ -0,0 +1,71 @@
+package com.owncloud.android.ui.dialog
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.owncloud.android.databinding.DialogFileAlreadyExistsBinding
+
+class FileAlreadyExistsDialog private constructor(
+    private val titleText: String?,
+    private val descriptionText: String?,
+    private val checkboxText: String?,
+    private var dialogClickListener: DialogButtonClickListener? = null
+) : DialogFragment() {
+
+    private lateinit var binding: DialogFileAlreadyExistsBinding
+    internal var isCheckBoxChecked: Boolean = false
+    internal var checkboxVisible: Boolean = true
+
+    interface DialogButtonClickListener {
+        fun onKeepBothButtonClick()
+        fun onSkipButtonClick()
+        fun onReplaceButtonClick()
+    }
+
+    data class Builder(
+        private var titleText: String? = null,
+        private var descriptionText: String? = null,
+        private var checkboxText: String? = null,
+        private var checkboxVisible: Boolean? = false,
+    ) {
+
+        fun setTitle(titleText: String) = apply { this.titleText = titleText }
+        fun setDescription(descriptionText: String) = apply { this.descriptionText = descriptionText }
+        fun setCheckboxText(checkboxText: String) = apply { this.checkboxText = checkboxText }
+        fun build() = FileAlreadyExistsDialog(titleText, descriptionText, checkboxText)
```

Comment:
If 3 or more parameter it's recommended to add the names.

Reply:
Done